recipes/graylog-forwarder/files/pre-install

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+set -e
+
+user="graylog-forwarder"
+group="graylog-forwarder"
+datadir="/var/lib/graylog-forwarder"
+
+case "$1" in
+	# DEB based systems
+	install|upgrade)
+		if ! getent group "$group" > /dev/null 2>&1 ; then
+			addgroup --system "$group" --quiet
+		fi
+
+		if ! id "$user" > /dev/null 2>&1 ; then
+			adduser --system --home "$datadir" --no-create-home \
+				--ingroup "$group" --disabled-password \
+				--gecos "Graylog Forwarder" \
+				--shell /bin/false --quiet "$user"
+		fi
+		;;
+	abort-deconfigure|abort-upgrade|abort-remove)
+		# Ignore
+		;;
+	# RPM based systems
+	1|2)
+		# Create graylog group
+		if ! getent group "$user" >/dev/null; then
+			groupadd -r "$user"
+		fi
+
+		# Create graylog user
+		if ! getent passwd "$user">/dev/null; then
+			useradd -r -M -g "$group" -d "$datadir" \
+				-s /sbin/nologin -c "Graylog Forwarder" "$user"
+		fi
+		;;
+	*)
+		echo "[ERROR] pre-install script called with unknown argument: '$1'"
+		exit 1
+		;;
+esac
+
+# Create directories
+install -d -o "$user" -g "$group" -m 0755 "$datadir"
```

Comment:
Let's also create the `data` directory in datadir:
```suggestion
install -d -o "$user" -g "$group" -m 0755 "$datadir"
install -d -o "$user" -g "$group" -m 0755 "$datadir/data"
```

Reply:
Okay.